### ISSUE TITLE
bug - Workload - the app filter provided as an example in the tracker search constraint can lead to the non detection of some use cases of execution errors

### ISSUE THREAD

**Issue by sebwurl** [trackme-limited/trackme-report-issues#151]
Hi,

I'm testing the Workload feature on v2.0.36 and created a simple scheduled report. It appears as an entity in the overview.
I changed my SPL to produce an error.
`index=_internal | head 1 | eval reason:error`

Since then, TrackMe did not recognize new executions (the number did not grow) and the entity is still showing a green status. Although, I can find the corresponding scheduler errors by clicking the drilldown from the entities detail page.

![image](https://github.com/trackme-limited/trackme-report-issues/assets/85566586/ac48967a-18fd-4de5-aa9d-c8b5e281ef74)

**Comment by guilhemmarchand** [comment]
Thanks @sebwurl 

Will be verified - it is possible that somehow this doesn't fit into an exact scheduler error (unlike for instance calling a non existing knowledge object)

Note that at the worst and since the execution will not happen any longer, the Workload should trigger a red status on the entity within an hour due to execution_delayed. (saying if we don't catch the scheduling errors we will catch the non execution normally)

Will be double checked.

**Comment by guilhemmarchand** [comment]
Note: A great test for instance is to have your scheduled search using a lookup call.
Then after some valid execution, either remove the lookup, or change its structure (to make the key or output fields unavailable) or modify the search to do a wrong call

**Comment by guilhemmarchand** [comment]
So @sebwurl thank you.

Unfortunate your first test didn't show up with the cool value that the Workload leverages.

Anyway, the issue is wit the "app" extraction, in the tracker generated, there is an "app=*" filter in the first pipe:

However, with this specific example of your search, Splunk does not extract the app field:

<img width="1783" alt="image" src="https://github.com/trackme-limited/trackme-report-issues/assets/7503059/73d9915e-25ee-4c92-b193-86de933234dd">

<img width="1783" alt="image" src="https://github.com/trackme-limited/trackme-report-issues/assets/7503059/69ee24e5-ed07-4023-8e85-483db11eabf1">

This was actually identified but later on we had added the app filter to allow the management of app at the discovery level.

Will be addressed in 2.0.38, but if you move the app=* to after:

```
| eval user=coalesce(user, user_alt), app=coalesce(app, app_alt), savedsearch_name=coalesce(savedsearch_name, savedsearch_name_alt)
```

or you remove it. this will capture this use case.

We will update the tracker creation to move the app filter

**Comment by sebwurl** [comment]
The entity did not end up in a "execution_delayed" status. It is still green, although the scheduler tries to run it every 5min. This is also related to the missing app field?

Regarding the user space: The user in the savedsearch_id is still correct in my error logs, it is still "admin" not "nobody".

Btw: I also tried to trigger the use case by disabling a macro of a another report. This entity is not in "error" status nor in "delay", it is shown as "skipped".

**Comment by guilhemmarchand** [comment]
Hi @sebwurl 

All these issues have been addressed in v2.0.38, there are as well some workload specific steps you will want to review in the release notes:

https://docs.trackme-solutions.com/releasenotes.html#version-2-0-38-build-1687154702-19-06-2023

Thank you
